src/main/resources/migration/changelog-v5.6.0.xml

```diff
@@ -107,4 +107,166 @@
+            </column>
+            <column name="NAME" type="VARCHAR(64)"/>
+            <column name="ALGORITHMS" type="TEXT[]"/>
+            <column name="IDENTIFIERS" type="TEXT[]"/>
+        </createTable>
+
+        <createTable tableName="PROTOCOL_IKEV2_TYPES">
+            <column name="PROTOCOL_PROPERTY_ID" type="BIGINT"/>
+            <column name="IKEV2_TYPE_ID" type="BIGINT"/>
+        </createTable>
+
+        <createTable tableName="IKEV2_TYPE">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="IKEV2_TYPE_PK"/>
+            </column>
+            <column name="TYPE" type="VARCHAR(16)"/>
+            <column name="REFS" type="TEXT[]"/>
+        </createTable>
+
+        <addForeignKeyConstraint baseTableName="COMPONENT" baseColumnNames="CRYPTO_PROPERTIES_ID"
+                                 constraintName="COMPONENT_CRYPTO_PROPERTIES_FK" deferrable="true" initiallyDeferred="true"
+                                 referencedTableName="CRYPTO_PROPERTIES" referencedColumnNames="ID"
+                                 onDelete="CASCADE" onUpdate="NO ACTION" validate="true"/>
+        <addForeignKeyConstraint baseTableName="CRYPTO_PROPERTIES" baseColumnNames="ALGORITHM_PROPERTIES_ID"
+                                 constraintName="CRYPTO_PROPERTIES_ALGORITHM_FK" deferrable="true" initiallyDeferred="true"
+                                 referencedTableName="ALGORITHM_PROPERTIES" referencedColumnNames="ID"
+                                 onDelete="CASCADE" onUpdate="NO ACTION" validate="true"/>
+        <addForeignKeyConstraint baseTableName="CRYPTO_PROPERTIES" baseColumnNames="CERTIFICATE_PROPERTIES_ID"
+                                 constraintName="CRYPTO_PROPERTIES_CERTIFICATE_FK" deferrable="true" initiallyDeferred="true"
+                                 referencedTableName="CERTIFICATE_PROPERTIES" referencedColumnNames="ID"
+                                 onDelete="CASCADE" onUpdate="NO ACTION" validate="true"/>
+        <addForeignKeyConstraint baseTableName="CRYPTO_PROPERTIES" baseColumnNames="RELATED_MATERIAL_PROPERTIES_ID"
+                                 constraintName="CRYPTO_PROPERTIES_RELATED_MATERIAL_FK" deferrable="true" initiallyDeferred="true"
+                                 referencedTableName="RELATED_CRYPTO_MATERIAL_PROPERTIES" referencedColumnNames="ID"
+                                 onDelete="CASCADE" onUpdate="NO ACTION" validate="true"/>
+        <addForeignKeyConstraint baseTableName="CRYPTO_PROPERTIES" baseColumnNames="PROTOCOL_PROPERTIES_ID"
+                                 constraintName="CRYPTO_PROPERTIES_PROTOCOL_FK" deferrable="true" initiallyDeferred="true"
+                                 referencedTableName="PROTOCOL_PROPERTIES" referencedColumnNames="ID"
+                                 onDelete="CASCADE" onUpdate="NO ACTION" validate="true"/>
```

Comment:
Just a note so we won't forget: Postgres doesn't create indexes on FKs automatically. We'll need to create indexes explicitly if we need them to support fast lookups.